### PR TITLE
[setup_state::03] Docs

### PR DESCRIPTION
**Pull request by kasperski95**
- [x] update to match 02

### CODE REVIEW

File: website/docs/tutorials/guides/testing.md [hunk 1]
```diff
@@ -188,6 +188,39 @@ If your IDE supports Cairo and doesn't know how to import `protostar`, add the f
...
 
+## `setup_state`
+Often while writing tests you have some setup work that needs to happen before tests run. The hook `setup_tmp_setup` can simplify your test functions and will speed up your tests. Use `tmp_state` to pass the information between `setup_state` and test functions as demonstrated on the example below:
+
+```cairo
+%lang starknet
+
+@view
+func setup_state():
+    %{ tmp_state.contract_a_address = deploy_contract("./tests/integration/testing_hooks/basic_contract.cairo").contract_address %}
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply website/docs/tutorials/guides/testing.md]: `context`

File: website/docs/tutorials/guides/testing.md [hunk 2]
```diff
@@ -188,6 +188,39 @@ If your IDE supports Cairo and doesn't know how to import `protostar`, add the f
 
 You can find all [assert signatures here](https://github.com/software-mansion/protostar/blob/master/cairo/protostar/asserts.cairo).
 
+## `__setup__`
+Often while writing tests you have some setup work that needs to happen before tests run. The hook `__setup__` can simplify and speed up your tests. Use `context` hint local to pass data from `__setup__` to test functions as demonstrated on the example below:
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/testing.md:192]:
```suggestion
Often while writing tests you have some setup work that needs to happen before tests run. The hook `__setup__` can simplify and speed up your tests. Use `context` variable to pass data from `__setup__` to test functions as demonstrated on the example below:
```

File: website/docs/tutorials/guides/testing.md [hunk 3]
```diff
@@ -438,7 +471,9 @@ class DeployedContract:
 ```
 Deploys a contract given a path relative to a Protostar project root. The section [Deploying contracts from tests](#deploying-contracts-from-tests) demonstrates a usage of this cheatcode.
 
-
+:::warning
+Deploying contract is a slow operation. If it's possible try using this cheatcode in the [`__setup__` hook](#__setup__).
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/testing.md:475]:
```suggestion
Deploying a contract is a slow operation. If it's possible try using this cheatcode in the [`__setup__` hook](#__setup__).
```

File: website/docs/tutorials/guides/testing.md [hunk 4]
```diff
@@ -188,6 +188,39 @@ If your IDE supports Cairo and doesn't know how to import `protostar`, add the f
...
+
+    return ()
+end
+```
+:::warning
+Hint local `context` can only store strings, integers and booleans.
+:::
+
+:::info
+Protostar executes `__setup__` only once per a [test suite](https://en.wikipedia.org/wiki/Test_suite). Then, for each test case Protostar copies the StarkNet state and `context` object.
```
Comment (MaksymilianDemitraszek) [inline website/docs/tutorials/guides/testing.md:218]: I think we don't need this one. We don't explain harder words ex. `StarkNet state`, and it's not our job. 
```suggestion
Protostar executes `__setup__` only once per a test suite. Then, for each test case Protostar copies the StarkNet state and `context` object.
```

**Review by MaksymilianDemitraszek — APPROVED** [review]

File: website/docs/tutorials/guides/testing.md (hunk 4 above)
Comment (kasperski95) [reply website/docs/tutorials/guides/testing.md:218]: We don't require technical background from the target audience, and we use `test suite` term in the CLI output, so it's worth to at least add a link. It doesn't change anything for the guys familiar with this term.
`pros` > `cons`

**Pull request merged**